File: 0904-fruit-into-baskets/0904-fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        prev = 0
        curr = 0
        start = 0
        res = 0
        for i in range(len(fruits)):
            if fruits[i] == fruits[prev] or fruits[prev] == fruits[curr]:
                curr,prev = i,curr
            elif fruits[i] != fruits[prev] and fruits[i] != fruits[curr]:
                curr,prev = i, curr
                start = prev
            res = max((i - start) + 1, res)
        return res
                

File: 0904-fruit-into-baskets/0904-fruit-into-baskets.js
/**
 * @param {number[]} fruits
 * @return {number}
 */
var totalFruit = function(fruits) {
        let  prev = curr = res = start = 0
        for (let i = 0; i < fruits.length; i++){
            if (fruits[i] == fruits[prev] || fruits[prev] == fruits[curr])
                [curr,prev] = [i,curr]
            else if(fruits[i] != fruits[prev] && fruits[i] != fruits[curr]){
                [curr,prev] = [i, curr]
                start = prev
            }
            res = Math.max((i - start) + 1, res)
        }
        return res

    
};